src/modelBasedFixers/apex-pmd-violations-fixer.ts

```diff
@@ -195,6 +191,55 @@ export class ApexPmdViolationsFixer implements vscode.CodeActionProvider {
         return updatedLines.join(lineEnding);
     }
 
+    private formatCode(
+        replaceCode: string,
+        document?: vscode.TextDocument,
+        range?: vscode.Range
+    ): string {
+        if (!document) {
+            return replaceCode;
+        }
+    
+        // Get the indentation of the first line in the range
+        const startLine = range && range.start.line > 0 ? range.start.line - 1 : 0;
+        const baseIndentation = this.getLineIndentation(document, startLine);
+        
+        // Split the replacement code into lines
+        const lines = replaceCode.split(/\r?\n/);
+        
+        // First, normalize the code by removing all existing indentation
+        const normalizedLines = lines.map(line => line.trimStart());
+        
+        let indentLevel = 0;
+
+        // Process each line to match the document's indentation style
+        const formattedLines = normalizedLines.map((line) => {
+            if (line.trim() === '') {
+                // Preserve empty lines without indentation
+                return '';
+            }
+            if (line.startsWith('}')) {
+                indentLevel = Math.max(0, indentLevel - 1);
+            }
+
+            const indentation = baseIndentation + '    '.repeat(indentLevel);
+
+            // Adjust for next lines, this line needs no changes
+            if (line.includes('{')) {
```

Comment:
It's technically possible to do something like `if (x) {System.debug('asdfasdfasdf');}` all on one line, so I think you want to do `line.endsWith('{'})` to make sure you properly handle that case.

Reply:
Ok, I knew edge cases like that are possible. I took care of } <some code> {, but now this. I'll take care of this.